test/Fencer/Server/Test.hs

```diff
@@ -133,17 +231,39 @@ withServer
 withServer =
   withResource createServer destroyServer
 
+-- | Get a unique port number that is not used by other tests. Needed since
+-- all tests run in parallel.
+getUniquePort :: IO Port
+getUniquePort = atomicModifyIORef' next (\port -> (succ port, port))
+  where
+    next :: IORef Port
+    next = unsafePerformIO (newIORef defaultGRPCPort)
+    {-# NOINLINE next #-}
+
+{-# NOINLINE getUniquePort #-}
```

Comment:
What would have happened if this no-inlining wasn't here?

Reply:
If `getUniquePort` gets inlined, the variable might be recreated every time `getUniquePort` is called. I'll add a comment.